nestkernel/synaptic_element.cpp

```diff
@@ -0,0 +1,271 @@
+  : eta( 0.1 )
+  , eps( 0.7 )
+{
+  name = "gaussian";
+}
+
+void
+nest::GrowthCurveGaussian::get( DictionaryDatum& d ) const
+{
+  def< std::string >( d, "growth_curve", name );
+  def< double_t >( d, "eps", eps );
+  def< double_t >( d, "eta", eta );
+}
+
+void
+nest::GrowthCurveGaussian::set( const DictionaryDatum& d )
+{
+  updateValue< double_t >( d, "eps", eps );
+  updateValue< double_t >( d, "eta", eta );
+}
+
+nest::double_t
+nest::GrowthCurveGaussian::update( double_t t,
+  double_t t_minus,
+  double_t Ca_minus,
+  double_t z_minus,
+  double_t tau_Ca,
+  double_t growth_rate ) const
+{
+  double_t lag, dz, zeta, xi, Ca, z_value;
+  const double_t h = Time::get_resolution().get_ms();
+
+  // Numerical integration from t_minus to t
+  // use standard forward Euler numerics
+  zeta = ( eta - eps ) / ( 2.0 * sqrt( log( 2.0 ) ) );
+  xi = ( eta + eps ) / 2.0;
+
+  z_value = z_minus;
+  Ca = Ca_minus;
+
+  for ( lag = t_minus; lag < ( t - h / 2.0 ); lag += h )
+  {
+    Ca = Ca - ( ( Ca / tau_Ca ) * h );
+    dz = h * growth_rate * ( 2.0 * exp( -pow( ( Ca - xi ) / zeta, 2 ) ) - 1.0 );
+    z_value = z_value + dz;
+  }
+
+  //  return z_value;
+  if ( z_value > 0 )
+  {
+    return z_value;
+  }
+  else
+    return 0.0;
+}
+
+
+/* ----------------------------------------------------------------
+* SynapticElement
+* Default constructors defining default parameters and state
+* ---------------------------------------------------------------- */
+
+nest::SynapticElement::SynapticElement()
+  : z( 0.0 )
+  , z_t_( 0.0 )
+  , z_connected_( 0 )
+  , continuous_( true )
+  , growth_rate_( 1.0 )
+  , tau_vacant_( 10.0 )
+  , growth_curve_( new GrowthCurveLinear )
+{
+}
+
+nest::SynapticElement::SynapticElement( const SynapticElement& se )
+  : z( se.z )
+  , z_t_( se.z_t_ )
+  , z_connected_( se.z_connected_ )
+  , continuous_( se.continuous_ )
+  , growth_rate_( se.growth_rate_ )
+  , tau_vacant_( se.tau_vacant_ )
+{
+  growth_curve_ = new_growth_curve( se.growth_curve_->get_name() );
+  DictionaryDatum gc_parameters = DictionaryDatum( new Dictionary );
+  se.get( gc_parameters );
+  growth_curve_->set( gc_parameters );
+}
+
+nest::SynapticElement& nest::SynapticElement::operator=( const SynapticElement& other )
+{
+  if ( this != &other )
+  {
+    // 1: allocate new memory and copy the elements
+    GrowthCurve* new_gc = new_growth_curve( other.growth_curve_->get_name() );
+    DictionaryDatum gc_parameters = DictionaryDatum( new Dictionary );
+
+    other.get( gc_parameters );
+    new_gc->set( gc_parameters );
+
+    delete growth_curve_;
+    growth_curve_ = new_gc;
+
+    z = other.z;
+    z_t_ = other.z_t_;
+    z_connected_ = other.z_connected_;
+    continuous_ = other.continuous_;
+    growth_rate_ = other.growth_rate_;
+    tau_vacant_ = other.tau_vacant_;
+  }
+  return *this;
+}
+
+nest::GrowthCurve*
+nest::SynapticElement::new_growth_curve( std::string name )
+{
+  if ( not name.compare( "gaussian" ) )
```

Comment:
Is it possible to use name objects also here? I presume that these values are passed through a dictionary?